src/app/dashboard/whitelist/WhiteListPage.js

```diff
@@ -0,0 +1,378 @@
+  state = {
+    page: 0,
+    editInvestorsShowing: false,
+    editInvestors: [],
+  }
+
+  componentWillMount () {
+    this.props.initialize()
+  }
+
+  handleChangePages = (pc: PageChanger) => {
+    this.props.updateListLength(pc.pageSize)
+    this.setState({
+      page: (pc.page - 1),
+    })
+  }
+
+  handleDatePicker = (picker: DatePickerType) => {
+    if (picker.length === 2){
+      this.setState({
+        page: 0, //reset to initial page , othewise it might refer to a page that doesnt exist //TODO: make sure this is true, im not sure it is
+      })
+      this.props.getWhitelist(picker[0],picker[1])
+    }
+  }
+
+  handleEditInvestors = (investors: Array<Object>) => {
+    let addresses = []
+    for (let i = 0; i < investors.length; i++) {
+      addresses.push(investors[i].cells[0].value)
+    }
+    this.setState({
+      editInvestorsShowing: true,
+      editInvestors: addresses,
+    })
+  }
+
+  handleRequestSubmit = () => {
+    this.props.editInvestors(this.state.editInvestors)
+    this.setState({
+      editInvestorsShowing: false,
+    })
+  }
+
+  handleRequestClose = () => {
+    this.setState({
+      editInvestorsShowing: false,
+    })
+  }
+
+  onHandleInvestorSubmit = () => {
+    this.props.singleSubmit()
+    return true  //needed for the component from carbon to work properly
+  }
+
+  //renders the list by making it date strings and splitting up in pages, at the start of the render function
+  paginationRendering () {
+    let paginatedArray = []
+    let investors = this.props.investors
+    let pageNum = this.state.page
+    let listLength = this.props.listLength
+    let startSlice = pageNum*listLength
+    let endSlice = ((pageNum+1)*listLength)
+    paginatedArray = investors.slice(startSlice, endSlice)
+    let stringifiedArray = []
+    for (let i = 0; i <paginatedArray.length; i++){
+      let csvRandomID: string = uuidv4()
+      let stringifyAdded = null
+      //QUESTION: is this the best way to get rid of the flow error? I think there must be a better way, as this code doesn't really do anything
+      //because when pulling from events, added will ALWAYS exist, but because we pass the type with a ? it brings flow error
+      //(because the time added is got from 'now' on blockchain, not from user input)
+      if (typeof(paginatedArray[i].added) === 'object'){
+        stringifyAdded = dateFormat(paginatedArray[i].added)
+      }
+      let stringifyFrom = dateFormat(paginatedArray[i].from)
+      let stringifyTo = dateFormat(paginatedArray[i].to)
+      let stringifyInvestor: EventData = {
+        id: csvRandomID,
+        address: paginatedArray[i].address,
+        added: stringifyAdded,
+        addedBy: paginatedArray[i].addedBy,
+        from: stringifyFrom,
+        to: stringifyTo,
+      }
+      stringifiedArray.push(stringifyInvestor)
+    }
+    return stringifiedArray
+  }
+
+  removeInvestor = (investors: Array<Object>) => {
```

Comment:
Investor type

Reply:
This should have been named differently by me. It is actually an object that is returned from the carbon data component , and then from that object I extract the investor addresses. So I have renamed it to 'dataTableRow', instead of 'investors'